src/ucp/wireup/select.c

```diff
@@ -678,6 +679,44 @@ ucp_wireup_tl_iface_latency(ucp_context_h context,
     }
 }
 
+static UCS_F_NOINLINE int
+ucp_wireup_check_path_index(ucp_wireup_lane_desc_t *lane_desc,
+                            ucp_lane_type_t lane_type, unsigned path_index)
+{
+    if (path_index == UCP_WIREUP_PATH_INDEX_UNDEFINED) {
+        goto out;
+    }
+
+    /* Path index is relevant if existing lane alredy configured for the same
+     * usage type (e.g. RMA or AM bandwidth lanes). */
+    if ((lane_desc->path_index != path_index) &&
+        (lane_desc->path_index != UCP_WIREUP_PATH_INDEX_UNDEFINED) &&
+        (lane_desc->lane_types & UCS_BIT(lane_type))) {
```

Comment:
what if we didn't check lane_type here?

Reply:
Then AM_BW lane can affect RMA_BW lane selection and vice versa, like follows:
1. lane is selected as AM_BW  lane N0, which is reflected in `lane_desc` 
2. During selection of RMA_BW lane, some lane matching the same characteristics is selected, but with local path N1. 
3. It will be added as a new lane because local path indexes do not match